### ISSUE TITLE
Rename sender/receiver/track stats

### ISSUE THREAD

**Issue by henbos** [w3c/webrtc-stats#298]
Largely copied from https://github.com/w3c/webrtc-stats/pull/273.

We have these hierarchies (and similar for audio and receive):

`RTCMediaStreamTrackStats <- RTCVideoStreamTrackStats <- RTCVideoSenderStats <- RTCLocalVideoStreamTrackStats`

In other words

`Track <- [Audio/Video]Track <- [Audio/Video][Sender/Receiver] <- Local[Audio/Video]Track`

It gets a bit confusing about going back and forth between track language and sender/receiver language.

The way I see it, it's sender/receiver all the way down. The RTCLocal[Audio/Video]StreamTrackStats are the only special case where now we're not talking about a sender, now it changes to an attachment of a track (it's not even a track-stat, since attaching the same track multiple times yields different attachment stats). Which really is a subview of the sender stats for the duration a track is attached.

Lacking a better word that can be used to refer to both a sender and a receiver than "SenderReceiver" (and avoiding confusion with RTCRtpTransceiver), I would name them:

```
RTCSenderReceiverStats (previously RTCMediaStreamTrackStats)
RTC[Audio/Video]SenderReceiverStats (RTC[Audio/Video]StreamTrackStats)
RTC[Audio/Video][Sender/Receiver]Stats (same name as before)
RTCSender[Audio/Video]TrackAttachmentStats (RTCLocal[Audio/Video]StreamTrackStats)
```
In other words

`SenderReceiver <- [Audio/Video]SenderReceiver <- [Audio/Video][Sender/Receiver] <- [Audio/Video]TrackAttachment`

Since "objectDeleted" is now a member of all stats and "detached" has been removed, there is no need for a separate dictionary for track attachments, so we can either reuse the same dictionary or keep it around for editorial purposes to make the distinction between sender/receiver stats and track attachment stats.

**Comment by alvestrand** [comment]
Department of inane bikeshedding:

What about "RTCMediaHandler" for the top level object?

Thus:
<pre>
RTCMediaHandlerStats
   RTCAudioHandlerStats
      RTCAudioSenderStats
      RTCAudioReceiverStats
  RTCVideoHandlerStats
      RtCVideoSenderStats
      RTCVideoReceiversStats
</pre>
Almost makes sense?

**Comment by henbos** [comment]
That makes sense.

Not sure this is just bikeshedding, if @taylor-b got confused by the new structure (https://github.com/w3c/webrtc-stats/issues/293#issuecomment-357074110) I think we should do a better job with the naming/explaining :)

**Comment by alvestrand** [comment]
@jan-ivar you suggested the original set.

**Comment by vr000m** [comment]
@karthikbr82 and @lennart-csio PTAL

**Comment by henbos** [comment]
If if #235 uses an event you might be able to calculate the "track" attachment stats from the sender/receiver stats

**Comment by jan-ivar** [comment]
@henbos Yes this was my point in the [slide](https://docs.google.com/presentation/d/1o-o5jZBLw3Py1OuenzWDkxDG6NigSmLHvGw5KemKWLw/edit#slide=id.g2d42e878ad_0_31) we didn't get to at the interim. It's not entirely trivial though:

```js
let senderStat = [...(await sender.getStats()).values()].find(s => s.type == “sender”);
let oldTracks = [...sender.getCompletedStats().values()].filter(s => s.type == “track”);

trackFramesSent = oldTracks.reduce((n, old) => n - old.framesSent, senderStat.framesSent);
```
